### PR TITLE
redraft configuring Astro

### PR DESCRIPTION
**Pull request by sarah11918**
#### Description (required)

Is there something before draft? Because this is barely a draft! :sweat_smile:

This page attempts to figure out what from a VERY old page is worth keeping, and what should be added for very basic "configuring Astro" advice... maybe more like "New project setup" so as not to confuse it with necessarily with just the config file. e.g.:

image

image

This page will live on the Start sidebar, along with specific guides for environment variables, TypeScript, the dev toolbar and editor setup etc. So, it does not need to be comprehensive.

But, it should be helpful to someone brand new to Astro with super basic things they might want to configure as soon as they have a new project.

This page also contains a random selection of other items which may or may not belong here? Let's fine out!

### CODE REVIEW

File: src/content/docs/en/guides/configuring-astro.mdx [hunk 1]
```diff
@@ -4,77 +4,46 @@ i18nReady: true
...
 ```js
 // astro.config.mjs
 import { defineConfig } from 'astro/config'
 
 export default defineConfig({
   // your configuration options here...
-  // https://docs.astro.build/en/reference/configuration-reference/
 })
 ```
+A JavaScript configuration file is included in every starter project created with our CLI wizard `create astro`. It is only required if you have something to configure, but most projects will use this file.
```
[the remark this answers is not in the document]
Comment (sarah11918) [reply src/content/docs/en/guides/configuring-astro.mdx:21]: That sounds great! The previous seemed a little heavy for a "quick set up stuff" page, but yours is great! 🙌

**Comment by ArmandPhilippot** [comment]
I know you asked for ideas during T&D, sorry... I didn't rush, because I honestly don't know what should be on this page. I would say it really depends on the project. What I do in all cases is:
1. Configuring extra tools like Prettier, ESlint, cspell, etc. (But, this is not Astro specific and this is already covered in Editor Setup)
2. In `astro.config.mjs`, the only thing I would see common to all projects is to configure `site`...
3. I make a file for my constants (this is not Astro specific, but it might be useful for people used to CMS, like the question you show coming from Discord):

```ts
export const CONFIG = {
    /* Because I can use it somewhere other than `<Head />` (header, footer...). */
    SITE_TITLE: "My awesome website!",
    /* I know this is the default with `paginate`, but if I want to change the value in multiple places it's easier. */
    ITEMS_PER_PAGE: 10
}
```
4. If I know I'm going to use some integrations like MDX or sitemap, I add them (although I usually add them later)...

Other than those points, I don't see. I think this is precisely one of the strong points of Astro, we have enough freedom to configure/structure our project as we wish.

So your updates seem fine to me! And I don't see what else we could add... Sorry for not being more helpful here. 😅

**Comment by sarah11918** [comment]
Thank you, @ArmandPhilippot ! Have also been discussing with Chris and it's OK if this is more of a landing/jumping off page to other types of configuration.

It really only needs to be a "HALP, I have no idea what to do first!" kind of page. 😄

File: src/content/docs/en/guides/configuring-astro.mdx [hunk 2]
```diff
@@ -1,131 +1,114 @@
...
+Every Astro starter project includes a `tsconfig.json` file in your project. Astro's [component script](/en/basics/astro-components/#the-component-script) is Typescript, which provides Astro's editor tooling and allows you to optionally add syntax to your JavaScript for type checking of your own project code.
 
-```bash
-# Example: Reads your configuration from ./astro.config.mjs
-astro build
-```
+Use the `tsconfig.json` file to configure the TypeScript template that will perform type checks on your code, configure TypeScript plugins, set import aliases, and more.
 
-You can explicitly set a config file to use with the `--config` CLI flag. This CLI flag always resolves relative to the current working directory where you ran the `astro` CLI command.
+<ReadMore>Read Astro's [TypeScript guide](/en/reference/configuration-reference/) for a full overview of TypeScript options and Astro's built-in utility types.</ReadMore>
```
Comment (ArmandPhilippot) [inline src/content/docs/en/guides/configuring-astro.mdx:40]: nit: I think this should be a link to Typescript, not Configuration reference:
```suggestion
<ReadMore>Read Astro's [TypeScript guide](/en/guides/typescript/) for a full overview of TypeScript options and Astro's built-in utility types.</ReadMore>
```

**Review by ArmandPhilippot — COMMENTED** [review]
I like the page structure, I think redirecting to other pages this way can be helpful! 🙌🏽 

However, I think you forgot to move some content. 😅 Currently "Add site metadata" is empty and the whole `<Head />` content is under `Customize output filenames`... and I can't provide a suggestion because some parts are not detected as a change...

Also a nit regarding a link!

**Comment by sarah11918** [comment]
D'oh! I did move stuff at the very end, yes. I'll clean that up. Thank you!

**Comment by delucis** [comment]
Er… oops, I must have accidentally discovered some keyboard shortcut that submits a review? Wasn’t done, but apparently off my first few comments wanted to go. Sorry about that!

File: src/content/docs/en/guides/configuring-astro.mdx [hunk 3]
```diff
@@ -1,131 +1,163 @@
...
+We recommend using the default file format `.mjs` in most cases, or `.ts` if you want to write TypeScript in your config file. However, `astro.config.js` and `astro.config.cjs` are also supported.
 
-## Supported Config File Types
 
-Astro supports several file formats for its JavaScript configuration file: `astro.config.js`, `astro.config.mjs`, `astro.config.cjs` and `astro.config.ts`. We recommend using `.mjs` in most cases or `.ts` if you want to write TypeScript in your config file.
+<ReadMore>Read Astro's [configuration reference](/en/reference/configuration-reference/) for a full overview of all supported configuration options.</ReadMore>
 
-TypeScript config file loading is handled using [`tsm`](https://github.com/lukeed/tsm) and will respect your project tsconfig options.
-## Config File Resolving
+## The TypeScript config File
```
Comment (delucis) [inline src/content/docs/en/guides/configuring-astro.mdx:34]: I might have been tempted to go even more minimal (e.g. link cards to the relevant docs for TS + editors) but these are great 👍

File: src/content/docs/en/guides/configuring-astro.mdx [hunk 4]
```diff
@@ -1,131 +1,163 @@
...
+<meta property="og:image" content="https://www.example.com/_astro/seo-banner.BZD7kegZ.webp">
+<meta property="og:image:alt" content="">
+
+<SomeOtherTags />
+
+<Favicon />
 ```
 
-Astro will resolve all other relative file and directory strings as relative to the project root:
+### Reference relative Files
```
Comment (delucis) [inline src/content/docs/en/guides/configuring-astro.mdx:121]: Neither this nor the “customize output filenames” examples are “common new project tasks” I don’t think. The filenames one is maybe a recipe? It should be a niche requirement (and one I’d recommend against unless someone _has_ to do this for some weird reason in their deploy setup). Definitely not something to recommend doing in a new project (and we’re mostly re-documenting Vite here, but necessarily so given it’s not so uncommon for someone to ask how to do this).

Referencing relative files also is _kind_ of niche, although admittedly very directly configuration-related. If the main usage is `root`, I might even be tempted to move this into the config reference for `root`…

File: src/content/docs/en/guides/configuring-astro.mdx [hunk 5]
```diff
@@ -1,131 +1,163 @@
...
   publicDir: new URL("./public", import.meta.url).toString(),
 })
 ```
 
 :::note 
-Vite-specific `import.meta` properties, like `import.meta.env` or `import.meta.glob`, are _not_ accessible from your configuration file. We recommend alternatives like [dotenv](https://github.com/motdotla/dotenv) or [fast-glob](https://github.com/mrmlnc/fast-glob) for these respective use cases. In addition, [tsconfig path aliases](https://www.typescriptlang.org/tsconfig#paths) will not be resolved. Use relative paths for module imports in this file.
+Vite-specific `import.meta` properties, like `import.meta.env` or `import.meta.glob`, are _not_ accessible from your configuration file. We recommend alternatives like [dotenv](https://github.com/motdotla/dotenv) or [fast-glob](https://github.com/mrmlnc/fast-glob) for these respective use cases.
+
+In addition, [tsconfig path aliases](https://www.typescriptlang.org/tsconfig#paths) will not be resolved. Use relative paths for module imports in this file.
 :::
```
Comment (delucis) [inline src/content/docs/en/guides/configuring-astro.mdx:154]: This note seems kind of irrelevant to this section? Whether we keep, move, or delete the section, we might want to delete this note.

**Review by delucis — COMMENTED** [review]
OK, couple of notes on the rest of the new configuration guide!

**Review by delucis — APPROVED** [review]
Looks great! Thank you @sarah11918 💜
